ksqldb-api-client/src/main/java/io/confluent/ksql/api/client/impl/DdlDmlResponseHandlers.java

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright 2020 Confluent Inc.
+ *
+ * Licensed under the Confluent Community License (the "License"); you may not use
+ * this file except in compliance with the License.  You may obtain a copy of the
+ * License at
+ *
+ * http://www.confluent.io/confluent-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package io.confluent.ksql.api.client.impl;
+
+import io.confluent.ksql.api.client.exception.KsqlClientException;
+import io.vertx.core.json.JsonObject;
+import java.util.concurrent.CompletableFuture;
+
+final class DdlDmlResponseHandlers {
+
+  public static final String EXECUTE_STATEMENT_REQUEST_ACCEPTED_DOC =
+      "The ksqlDB server accepted the statement issued via executeStatement(), but the response "
+      + "received is of an unexpected format. ";
+  public static final String EXECUTE_STATEMENT_USAGE_DOC = "The executeStatement() method is only "
+      + "for 'CREATE', 'CREATE ... AS SELECT', 'DROP', 'TERMINATE', and 'INSERT INTO ... AS "
+      + "SELECT' statements. ";
+
+  private DdlDmlResponseHandlers() {
+  }
+
+  static void handleExecuteStatementResponse(
+      final JsonObject ksqlEntity,
+      final CompletableFuture<Void> cf
+  ) {
+    if (!isCommandStatusEntity(ksqlEntity)) {
+      handleUnexpectedEntity(ksqlEntity, cf);
+      return;
+    }
+
+    cf.complete(null);
+  }
+
+  static RuntimeException handleUnexpectedNumResponseEntities(final int numEntities) {
+    if (numEntities == 0) {
+      return new KsqlClientException(EXECUTE_STATEMENT_REQUEST_ACCEPTED_DOC
+          + EXECUTE_STATEMENT_USAGE_DOC);
+    }
+
+    throw new IllegalStateException(
+        "Unexpected number of entities in server response: " + numEntities);
+  }
+
+  private static boolean isCommandStatusEntity(final JsonObject ksqlEntity) {
+    return ksqlEntity.getString("commandId") != null
+        && ksqlEntity.getJsonObject("commandStatus") != null;
+  }
+
+  // CHECKSTYLE_RULES.OFF: CyclomaticComplexity
+  private static void handleUnexpectedEntity(
```

Comment:
This seems fair enough, but another way of doing this would be to check the sql the user passes to executeStatement, if it's not one of the allowed commands we can reject it before sending to the server. Would just need to trim and lower case the string, then check for string prefix.

Reply:
I agree the approach implemented here and the alternative approach of checking for statement prefixes are about equally brittle. The reason I prefer the implemented approach is because if we reject statements outright (before sending them to the server), then we should reject all non-DDL/DML statements outright, which requires a more complicated check than just a simple prefix check. The check would have to be:
- starts with `CREATE STREAM` or `CREATE TABLE`, or
- starts with `DROP` and does not contain `CONNECTOR` (and special casing in case connector is in quotes), or
- starts with `TERMINATE`, or
- starts with `INSERT INTO` and contains `AS SELECT` (in order to distinguish from `INSERT INTO ... VALUES` statements)

This logic is complicated enough that I think we'd be entering fuzzy territory on whether the client or the server should perform parsing/validation. 

However, there's definitely a case to be made that we should perform this more complicated check on the client, in order to prevent users from using `executeStatement()` to insert rows or create/drop connectors. I just don't think it makes sense to have a partially-fledged prefix check on the statement -- we should either go all in or not bother, haha.